semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -3056,6 +3057,85 @@ and m_directive a b =
   | G.OtherDirective _ ->
       fail ()
 
+and m_directive_vs_def a b =
+  let f (filea, importsa) = function
+    (* const x = require('y');
```

Comment:
Does it have to be specifically `const` or is that just an example?

Reply:
That's just an example. I will clarify.